src/Npgsql/TypeHandlers/CompositeHandlers/CompositeParameterHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Diagnostics;
+using System.Reflection;
+using System.Threading.Tasks;
+using Npgsql.TypeHandling;
+
+namespace Npgsql.TypeHandlers.CompositeHandlers
+{
+    abstract class CompositeParameterHandler
```

Comment:
Consider moving this (and the generic subclass) into CompositeHandler as an inner type, it's really an implementation detail.

Reply:
I don't want to have a super large file (if in case I do it, then parameter handlers should be moved to). There is a special folder for all composite stuff (:

Reply:
Another option is to have several files for CompositeHandler, using partial classes. A namespace with several types is also OK.